src/modules/worker/worker.service.ts

```diff
@@ -1,24 +1,120 @@
 import { Injectable, OnModuleInit } from '@nestjs/common';
-// @ts-expect-error - @y/redis is only having jsdoc types
-import { createWorker } from '@y/redis';
+import { RedisAdapter } from 'infra/redis/redis.adapter.js';
+import { randomUUID } from 'node:crypto';
+import { Logger } from '../../infra/logger/index.js';
+import { Task } from '../../infra/redis/interfaces/redis.interface.js';
 import { RedisService } from '../../infra/redis/redis.service.js';
 import { StorageService } from '../../infra/storage/storage.service.js';
+import { Api, createApiClient } from '../../infra/y-redis/api.service.js';
+import { decodeRedisRoomStreamName } from '../../infra/y-redis/helper.js';
 import { WorkerConfig } from './worker.config.js';
 
 @Injectable()
 export class WorkerService implements OnModuleInit {
+	private client!: Api;
+	private readonly consumerId = randomUUID();
+	private redis!: RedisAdapter;
+
 	public constructor(
-		private readonly storage: StorageService,
+		private readonly storageService: StorageService,
 		private readonly redisService: RedisService,
+		private readonly logger: Logger,
 		private readonly config: WorkerConfig,
-	) {}
+	) {
+		this.logger.setContext(WorkerService.name);
+	}
 
 	public async onModuleInit(): Promise<void> {
-		await createWorker(
-			await this.storage.get(),
-			this.config.REDIS_PREFIX,
-			{},
-			this.redisService.createRedisInstance.bind(this.redisService),
+		this.client = await createApiClient(this.storageService, this.redisService);
+		this.redis = await this.redisService.createRedisInstance();
+
+		this.logger.log(`Created worker process ${this.consumerId}`);
+		while (!this.client._destroyed) {
+			await this.consumeWorkerQueue();
+		}
+		this.logger.log(`Ended worker process ${this.consumerId}`);
+	}
+
+	public async consumeWorkerQueue(): Promise<Task[]> {
+		const tryClaimCount = this.config.WORKER_TRY_CLAIM_COUNT;
+		const taskDebounce = this.config.WORKER_TASK_DEBOUNCE;
+		const minMessageLifetime = this.config.WORKER_MIN_MESSAGE_LIFETIME;
+		const tasks: Task[] = [];
+
+		const reclaimedTasks = await this.redis.reclaimTasks(this.consumerId, taskDebounce, tryClaimCount);
+
+		const deletedDocEntries = await this.redis.getDeletedDocEntries();
+		const deletedDocNames = deletedDocEntries?.map((entry) => {
+			return entry.message.docName;
+		});
+
+		reclaimedTasks.messages?.forEach((m) => {
+			const stream = m?.message.compact;
+			stream && tasks.push({ stream: stream.toString(), id: m?.id.toString() });
+		});
+		if (tasks.length === 0) {
+			this.logger.log(`No tasks available, pausing... ${JSON.stringify({ tasks })}`);
+			await new Promise((resolve) => setTimeout(resolve, 1000));
+
+			return [];
+		}
+
+		this.logger.log(`Accepted tasks ${JSON.stringify({ tasks })}`);
+
+		await Promise.all(
+			tasks.map(async (task) => {
+				const streamlen = await this.redis.tryClearTask(task);
```

Comment:
Die Funktion innerhalb des maps in eine extra Methode. 

const XXX = tasks.map(this.YYY(task));
await Promise.all(XXX);